packages/core/src/types/utils/merkle-tree.interface.ts

```diff
@@ -0,0 +1,56 @@
+import { BigNumber } from '../number'
+
+export interface MerkleTreeNode {
+  key: BigNumber
```

Comment:
Just to confirm--this is both the depth and position of the node in the tree?

Reply:
No. It does not contain depth information. Bits after the bit that indicates whether this node is the left or right child of its parent will exist but will all be zeros. This makes it equivalent to all keys for all left descendants of this node as well.

Reply:
Alternatively, we could right-shift it so that the least significant bit indicates whether this node is the left or right child of its parent, but 1) we'd have to left shift it back to actually walk the tree and 2) we'd have the same problem in that the key for 0 is the same as 00, which is the same as 000 in that they'd all just be the `BigNumber(0)`

Reply:
Got it--I trust your approach, was just trying to better understand what the "key" of a non-leaf node is. 😃

Reply:
Ohh, gotcha. It's just used to de-dupe in storage. Before ID in the DB was just the hash, but that wasn't sufficient because if we stored two intermediate nodes at different spots in the tree that happen to have the same hash, we don't want an update to one to delete the other and vice-versa. This solves that: https://github.com/willmeister/pigi/blob/feat/386/MerkleInclusionQuantifier/packages/core/src/app/utils/merkle-tree.ts#L556

Reply:
Right, I remembered this convo and that's why I assumed it was depth and position--but that solution makes sense.